### PR TITLE
Add http4s module

### PR DESCRIPTION
**Pull request by jcranky**
Very simple for now, but adds the whole skeleton in case someone needs something more.

### CODE REVIEW

**Comment by leifwickland** [comment]
@jcranky, Thanks for the PR! Aside from the broken 2.10 build, it looks good to me.

The 2.10 build is failing because http4s isn't published for that version of scala. I don't recall how to exclude your new sbt subproject from being built for that part of the cross build. Do you?

**Comment by jcranky** [comment]
I'm not sure, but I'll push a commit trying something in a minute. Lets see what happens :)

**Comment by leifwickland** [comment]
@jcranky I'm really baffled how that worked. Don't get me wrong: I'm glad it did, but the 2.10 build seems to have run the same course as the other builds, but succeeded this time.

**Comment by jcranky** [comment]
What do you mean? The http4s module is not there in the 2.10 build anymore, which makes sense to me :) - I was just not sure if that line I added would have the overriding behaviour I wanted, but it did :D

**Comment by leifwickland** [comment]
@jcranky Oh, gosh. My bad. Please pardon my pre-coffee lack of awareness.

Thanks again.

File: modules/http4s/build.sbt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name := "pureconfig-http4s"
+
+libraryDependencies ++= Seq(
+  "org.http4s" %% "http4s-core" % "0.18.0",
+  Dependencies.scalaMacrosParadise,
+  Dependencies.scalaTest
+)
+
+crossScalaVersions := Seq("2.11.11", "2.12.4")
```
Comment (ruippeixotog) [inline modules/http4s/build.sbt]: Even though this works perfectly, it is not so convenient when we have to update Scala versions (with this we'd have to update in multiple places). Can you use something like we did with [the Akka module](https://github.com/pureconfig/pureconfig/blob/master/modules/akka/build.sbt#L4) instead?

File: modules/http4s/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+
+# Http4s module for PureConfig
```
Comment (ruippeixotog) [inline modules/http4s/README.md:2]: I wonder if we should add something here saying that only `Uri` is supported for now. People may come here expecting readers for many HTTP model classes and find out that it's not the case. I'm ok with leaving this as it is, though :)

**Review by ruippeixotog — CHANGES_REQUESTED** [review]
Thanks for the new module @jcranky, this looks good :) I hope that readers for more HTTP model classes get added with time.

File: modules/http4s/build.sbt (hunk 1 above)
Comment (jcranky) [reply modules/http4s/build.sbt]: Ah, of course, that is better, will do.

File: modules/http4s/README.md (hunk 2 above)
Comment (jcranky) [reply modules/http4s/README.md:2]: Well it wouldn't hurt to do that, so I'll do it :)

**Comment by jcranky** [comment]
@ruippeixotog done

**Review by ruippeixotog — APPROVED** [review]
LGTM now. Thanks again!

**Comment by leifwickland** [comment]
@jcranky, I manually merged this branch into master in 9e21c48 to work around the failing Travis check.

I'd like to add you to the CONTRIBUTORS file. If you agree, please make a PR with a single line with your name, username and your email and I'll merge it.

Thank you for the contribution!

**Comment by jcranky** [comment]
thanks :)

**Pull request closed without merging**